tests/Unit/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/Test_BjorhusImpl.cpp

```diff
@@ -0,0 +1,157 @@
+      local_char_speeds.at(0)[i] = -0.3;
+      local_char_speeds.at(1)[i] = -0.1;
+    }
+  }
+
+  // Allocate memory for output
+  tnsr::aa<DataVector, VolumeDim, frame> local_bc_dt_v_psi(
+      slice_grid_points, std::numeric_limits<double>::signaling_NaN());
+
+  // Compute rhs value
+  GeneralizedHarmonic::BoundaryConditions::detail::
+      set_dt_v_psi_constraint_preserving(
+          make_not_null(&local_bc_dt_v_psi), local_unit_interface_normal_vector,
+          local_three_index_constraint, local_char_projected_rhs_dt_v_psi,
+          local_char_speeds);
+
+  // Initialize with values from SpEC
+  auto spec_bc_dt_v_psi =
+      make_with_value<tnsr::aa<DataVector, VolumeDim, frame>>(
+          local_bc_dt_v_psi, std::numeric_limits<double>::signaling_NaN());
+
+  for (size_t i = 0; i < slice_grid_points; ++i) {
+    get<0, 0>(spec_bc_dt_v_psi)[i] = 25.4;
+    get<0, 1>(spec_bc_dt_v_psi)[i] = 25.4;
+    get<0, 2>(spec_bc_dt_v_psi)[i] = 25.4;
+    get<0, 3>(spec_bc_dt_v_psi)[i] = 25.4;
+    get<1, 1>(spec_bc_dt_v_psi)[i] = 31.4;
+    get<1, 2>(spec_bc_dt_v_psi)[i] = 31.4;
+    get<1, 3>(spec_bc_dt_v_psi)[i] = 31.4;
+    get<2, 2>(spec_bc_dt_v_psi)[i] = 33.4;
+    get<2, 3>(spec_bc_dt_v_psi)[i] = 33.4;
+    get<3, 3>(spec_bc_dt_v_psi)[i] = 39.4;
+  }
+
+  // Compare values returned by BC action vs those from SpEC
+  CHECK_ITERABLE_APPROX(local_bc_dt_v_psi, spec_bc_dt_v_psi);
+
+  // Test for another set of values
+  {
+    // Setting unit_interface_normal_Vector
+    for (size_t i = 0; i < slice_grid_points; ++i) {
+      get<0>(local_unit_interface_normal_vector)[i] = -1.;
+      get<1>(local_unit_interface_normal_vector)[i] = 1.;
+      get<2>(local_unit_interface_normal_vector)[i] = 1.;
+    }
+  }
+  // Compute rhs value
+  GeneralizedHarmonic::BoundaryConditions::detail::
+      set_dt_v_psi_constraint_preserving(
+          make_not_null(&local_bc_dt_v_psi), local_unit_interface_normal_vector,
+          local_three_index_constraint, local_char_projected_rhs_dt_v_psi,
+          local_char_speeds);
+
+  // Initialize with values from SpEC
+  for (size_t i = 0; i < slice_grid_points; ++i) {
+    get<0, 0>(spec_bc_dt_v_psi)[i] = 20.;
+    get<0, 1>(spec_bc_dt_v_psi)[i] = 20.;
+    get<0, 2>(spec_bc_dt_v_psi)[i] = 20.;
+    get<0, 3>(spec_bc_dt_v_psi)[i] = 20.;
+    get<1, 1>(spec_bc_dt_v_psi)[i] = 26.;
+    get<1, 2>(spec_bc_dt_v_psi)[i] = 26.;
+    get<1, 3>(spec_bc_dt_v_psi)[i] = 26.;
+    get<2, 2>(spec_bc_dt_v_psi)[i] = 28.;
+    get<2, 3>(spec_bc_dt_v_psi)[i] = 28.;
+    get<3, 3>(spec_bc_dt_v_psi)[i] = 34.;
+  }
+
+  // Compare values returned by BC action vs those from SpEC
+  CHECK_ITERABLE_APPROX(local_bc_dt_v_psi, spec_bc_dt_v_psi);
+}
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.Evolution.Systems.GeneralizedHarmonic.BCBjorhus.VPsi",
+                  "[Unit][Evolution]") {
+  // Piece-wise tests with SpEC output
+  const size_t grid_size = 3;
+
+  test_constraint_preserving_bjorhus_v_psi(grid_size);
```

Comment:
Besides spec test, please add a python random number test unless that would be unfeasible for some reason

Reply:
Added